server/api/src/rsDataObjOpen.cpp

```diff
@@ -847,92 +850,96 @@ namespace
                 cond_input[OPEN_TYPE_KW] = std::to_string(OPEN_FOR_WRITE_TYPE);
             }
 
+            // The data object information pointer must be populated in order to do anything below.
             if (!info_head) {
                 THROW(SYS_REPLICA_DOES_NOT_EXIST, fmt::format(
                     "[{}] - no data object was found for [{}]",
                     __FUNCTION__, dataObjInp->objPath));
             }
 
-            irods::replica_state_table::instance().insert(*info_head);
+            // If the winning replica is not found in the list of replicas, something has gone horribly
+            // wrong and we should bail immediately. We need to reference the winning replica whereas in
+            // the past the linked list was sorted with the winning replica at the head.
+            auto obj = data_object::make_data_object_proxy(*info_head);
+            auto maybe_replica = data_object::find_replica(obj, hierarchy);
+            if (!maybe_replica) {
+                THROW(SYS_REPLICA_DOES_NOT_EXIST, fmt::format(
+                    "[{}] - no replica found for [{}] on [{}]",
+                    __FUNCTION__, obj.logical_path(), hierarchy));
+            }
+
+            auto replica = *maybe_replica;
 
-            // sort replica list based on some set of criteria
-            const auto open_for_write = getWriteFlag(dataObjInp->openFlags);
-            if (const int ec = sortObjInfoForOpen(&info_head, cond_input.get(), open_for_write); ec < 0) {
-                THROW(ec, fmt::format(
-                    "[{}] - Unable to select a data obj info matching the resource hierarchy from the keywords.",
-                    __FUNCTION__));
+            // We need to migrate data to the cache directory in the case of bundled data before we do
+            // anything else because this changes the data object information that we are working with.
+            if (leaf_resource_is_bundleresc(hierarchy)) {
+                if (const int ec = stage_bundled_data_to_cache_directory(rsComm, &info_head); ec < 0) {
+                    if (lockFd > 0) {
+                        unlock_data_obj();
+                    }
+                    freeAllDataObjInfo(info_head);
+                    return ec;
+                }
+
+                if (replica.get() != info_head) {
+                    replica = replica::make_replica_proxy(*info_head);
+                }
             }
 
-            apply_static_pep_data_obj_open_pre(*rsComm, *dataObjInp, &info_head);
+            // Record the current replica status so that it can be restored later.
+            const auto old_replica_status = replica.replica_status();
```

Comment:
'original_replica_status' is used in the log message on 910... ?   rather than old?   also referred to in the comments as 'before'... just working on consistency.